### PR TITLE
Aggregate extern(C++) changes into a single, more-detailed changelog.

### PR DESCRIPTION
**Pull request by TurkeyMan**
I thought the release notes should make a better deal of these changes. They're massive for users interacting with C++ like myself.

Depends on Rainer's final patch in this suite of work being merged: https://github.com/dlang/dmd/pull/8362

Ideally this should land in 2.081

### CODE REVIEW

**Comment by wilzbach** [comment]
> Ideally this should land in 2.081

-> Rebased to `stable`.

File: changelog/extern_cpp_overhaul.dd [hunk 1]
```diff
@@ -0,0 +1,24 @@
+`extern (C++)` construction, destruction, operators and other mangling improvements
+
+Many improvements have been made to the `extern(C++)` experience in this release cycle.
+
+Mangling improvements include:
+* Constructor/destructor mangling matches C++
+* Compatible D style operators now mangle as C++ operators
+* `nullptr_t` mangles correctly
+* Also various mangling bugs have been fixed
```
Comment (wilzbach) [inline changelog/extern_cpp_overhaul.dd]: This isn't Markdown, it's DDoc which for the time being doesn't support Markdown.

**Comment by TurkeyMan** [comment]
Depends on Rainer's patch, otherwise the text is a lie ;)

File: changelog/extern_cpp_overhaul.dd [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+`extern(C++)` APIs that use `nullptr_t` can use `typeof(null)` on the D side:
+---
+alias nullptr_t = typeof(null);
+extern (C++) void fun(nullptr_t);
```
Comment (wilzbach) [inline changelog/extern_cpp_overhaul.dd]: Maybe use `extern(C++)` for the sake of consistency? (also applies to some latter usages)

File: changelog/extern_cpp_overhaul.dd (hunk 1 above)
Comment (wilzbach) [reply changelog/extern_cpp_overhaul.dd]: I fixed this for you.

File: changelog/extern_cpp_overhaul.dd [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+Many improvements have been made to the `extern(C++)` experience in this release cycle.
+
+Mangling improvements include:
+
+$(UL
+  $(LI  Constructor/destructor mangling matches C++)
+  $(LI  Compatible D style operators now mangle as C++ operators)
+  $(LI  `nullptr_t` mangles correctly)
+  $(LI  Also various mangling bugs have been fixed)
```
Comment (Geod24) [inline changelog/extern_cpp_overhaul.dd]: I would leave the `Also` out.

**Comment by TurkeyMan** [comment]
Rainer's patch was merged, so this can too.

**Pull request merged**